0.preprocess-sites/2.process-cells.py

```diff
@@ -0,0 +1,168 @@
+# 0.prefilter-features.py must be run first
+try:
+    all_feature_df = pd.read_csv(prefilter_file, sep="\t")
+except FileNotFoundError:
+    raise FileNotFoundError(
+        "Error",
+        f"{prefilter_file} not found.  ",
+        "Perform 0.prefilter-features.py prefilter before continuing..."
+    )
+
+sites = [x for x in os.listdir(batch_dir) if x not in ignore_files]
+
+for site in sites:
+    try:
+        print(f"Now processing {site}...")
+        output_folder = pathlib.Path(output_basedir, batch, "paint", site)
+        os.makedirs(output_folder, exist_ok=True)
+
+        meta_output_file = pathlib.Path(output_folder, f"metadata_{site}.tsv.gz")
+        count_output_file = pathlib.Path(output_folder,f"cell_counts_{site}.tsv")
+
+        compartment_dir = pathlib.Path(batch_dir, site)
+
+        #Make the compartment_csvs dictionary used to merge dfs
+        compartment_csvs = {}
+        for compartment in compartments:
+            try:
+                metadata_cols = parent_cols[compartment.lower()] + id_cols
+            except KeyError:
+                metadata_cols = id_cols
+            compartment_csvs[compartment] = load_single_cell_compartment_csv(
+                compartment_dir, compartment, metadata_cols
+            )
+
+        #Load .tsv created in 1.process-spots
+        foci_file = pathlib.Path(foci_dir, site, "full_cell_category_scores_by_guide.tsv.gz")
+        try:
+            foci_df = pd.read_csv(foci_file, sep="\t")
+        except FileNotFoundError:
+            print("""Run 1.process-spots before continuing. \n
+                  If it has been run, check that output_basedir is set correctly in
+                  the process-spots section of the config""")
+
+        #Relabel columns in foci_df to start with Metadata_Foci_
+        foci_df.columns = ["Metadata_Foci_{}".format(x) for x in foci_df.columns]
+
+    except FileNotFoundError:
+        print(f"{site} data not found")
+        continue
+
+
+    # Merge compartment csvs
+    sc_merged_df = merge_single_cell_compartments(compartment_csvs, merge_info, id_cols)
+    sc_merged_df = sc_merged_df.sort_values(by=sort_col)
+
+    # Add metadata
+    sc_merged_df = sc_merged_df.rename({"Metadata_Cells_ImageNumber": "Metadata_ImageNumber"}, axis="columns")
+    sc_merged_with_metadata_df = foci_df.merge(
+        sc_merged_df,
+        left_on=metadata_merge_left,
+        right_on=metadata_merge_right,
+        how="right",
+    )
+
+    metadata_df = sc_merged_with_metadata_df.loc[
+        :, sc_merged_with_metadata_df.columns.str.contains("Metadata")
+    ].drop_duplicates()
+
+    metadata_df.loc[
+    metadata_df.Metadata_Foci_Cell_Category.isna(), "Metadata_Foci_Cell_Category"
+    ] = 5
+    metadata_df.Metadata_Foci_Cell_Category = metadata_df.Metadata_Foci_Cell_Category.astype(
+        int
+    )
+    metadata_df.loc[metadata_df.Metadata_Foci_site.isna(), "Metadata_Foci_site"] = site
```

Comment:
`Metadata_Foci_site` is NA when cells don't have foci. But the site still exists and shouldn't be NA.